it/src/main/java/com/google/cloud/teleport/it/bigtable/StaticBigtableResourceManager.java

```diff
@@ -0,0 +1,284 @@
+
+    // create the bigtable admin and data client settings builders, and set the necessary id's for
+    // each.
+    BigtableInstanceAdminSettings.Builder bigtableInstanceAdminSettings =
+        BigtableInstanceAdminSettings.newBuilder().setProjectId(builder.projectId);
+    BigtableTableAdminSettings.Builder bigtableTableAdminSettings =
+        BigtableTableAdminSettings.newBuilder()
+            .setProjectId(builder.projectId)
+            .setInstanceId(this.instanceId);
+    BigtableDataSettings.Builder bigtableDataSettings =
+        BigtableDataSettings.newBuilder()
+            .setProjectId(builder.projectId)
+            .setInstanceId(this.instanceId);
+
+    // add the credentials to the builders, if set.
+    if (builder.credentialsProvider != null) {
+      bigtableInstanceAdminSettings.setCredentialsProvider(builder.credentialsProvider);
+      bigtableTableAdminSettings.setCredentialsProvider(builder.credentialsProvider);
+      bigtableDataSettings.setCredentialsProvider(builder.credentialsProvider);
+    }
+
+    this.projectId = builder.projectId;
+    this.bigtableResourceManagerClientFactory =
+        new BigtableResourceManagerClientFactory(
+            bigtableInstanceAdminSettings.build(),
+            bigtableTableAdminSettings.build(),
+            bigtableDataSettings.build());
+  }
+
+  public static StaticBigtableResourceManager.Builder builder(String projectId) throws IOException {
+    return new StaticBigtableResourceManager.Builder(projectId);
+  }
+
+  /**
+   * Returns the project ID this Resource Manager is configured to operate on.
+   *
+   * @return the project ID.
+   */
+  public String getProjectId() {
+    return projectId;
+  }
+
+  /**
+   * Return the instance ID this Resource Manager uses to create and manage tables in.
+   *
+   * @return the instance ID.
+   */
+  public String getInstanceId() {
+    return instanceId;
+  }
+
+  public String getTableId() {
+    return tableId;
+  }
+
+  @Override
+  public synchronized void createInstance(Iterable<BigtableResourceManagerCluster> clusters) {
+    LOG.info(
+        "No Cloud Bigtable instances are going to be created. Using static instance: {}",
+        instanceId);
+  }
+
+  /**
+   * Helper method for determining if the given tableId exists in the instance.
+   *
+   * @param tableId The id of the table to check.
+   * @throws IllegalStateException if the table does not exist in the instance.
+   */
+  private void checkHasTable(String tableId) {
+    try (BigtableTableAdminClient tableAdminClient =
+        bigtableResourceManagerClientFactory.bigtableTableAdminClient()) {
+      if (!tableAdminClient.exists(tableId)) {
+        throw new IllegalStateException(
+            "The table " + tableId + " does not exist in instance " + instanceId + ".");
+      }
+    }
+  }
+
+  @Override
+  public synchronized void createTable(String newTableId, BigtableTableSpec spec) {
+    if (!Objects.equals(tableId, newTableId)) {
+      LOG.warn(
```

Comment:
Should this throw an exception? not sure exactly how ResourceMananger is used but seems like we should fail the test here since it would never be able to use newTableId

Reply:
Changed to throw BigtableResourceManagerException